src/buffer.rs

```diff
@@ -593,15 +593,12 @@ mod tests {
 
         let inputs = vec![in1.as_ptr(), in2.as_ptr()];
         let mut outputs = vec![out1.as_mut_ptr(), out2.as_mut_ptr()];
-        let mut buffer =
-            unsafe { AudioBuffer::from_raw(2, 2, inputs.as_ptr(), outputs.as_mut_ptr(), SIZE) };
+        let mut buffer = unsafe { AudioBuffer::from_raw(2, 2, inputs.as_ptr(), outputs.as_mut_ptr(), SIZE) };
 
         for (input, output) in buffer.zip() {
-            input.into_iter().zip(output.into_iter()).fold(0, |acc,
-             (input,
-              output)| {
-                assert_eq!(*input - acc as f32, 0.0);
-                assert_eq!(*output, 0.0);
+            input.iter().zip(output.iter_mut()).fold(0, |acc, (input, output)| {
+                test_util::assert_f32_equal(*input, acc as f32);
+                test_util::assert_f32_zero(*output);
```

Comment:
The output was set to 0.0 explicitly, and the assert checks that it was not changed. It would seem that an exact comparison is more appropriate here.